### PR TITLE
chore: bump Python version, update poetry, update deps, fix tests

### PR DESCRIPTION
**Pull request by francescofuggitti**
## Proposed changes

This is a sub-PR to #81 due to failing tests

- Bump Python version to 3.10
- Update poetry deps

## Fixes

- fix tests

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/master/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/grammars.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ FALSE.2: /false/
 
 // Symbols cannot contain uppercase letters, because these are reserved
 // Moreover, any word between quotes is a symbol
-SYMBOL_NAME: /[a-z][a-z0-9_]*|"\w+"/
+SYMBOL_NAME: /[a-z][a-z0-9_-]*[a-z0-9_]*|"\w+"/
```
Comment (marcofavorito) [inline docs/grammars.md:42]: this does not look right.

See e.g. here: https://regex101.com/r/NYgppM/1 ; `a-` is accepted.

Comment (francescofuggitti) [reply docs/grammars.md:42]: True! It was not that a simple fix ;)

What about this one? 
`^(?:"[^"]*"|(?:[a-z][\w-]*[a-z]|[a-z]))$`

File: tox.ini [hunk 2]
```diff
@@ -1,135 +1,129 @@
 [tox]
 isolated_build = True
-envlist = bandit, safety, check-copyright, black-check, isort-check, vulture, flake8, mypy, pylint, py3.8, py3.7, docs
+envlist = bandit, safety, check-copyright, black-check, isort-check, vulture, flake8, mypy, docs, py3{8,9,10,11}
```
Comment (francescofuggitti) [inline tox.ini:3]:
```suggestion
envlist = bandit, safety, check-copyright, black-check, isort-check, vulture, flake8, mypy, docs, pylint, py3{8,9,10,11}
```

Pylint gave lots of errors, so I deleted it for the moment. However, we might want to fix them all and have it here again.

Comment (marcofavorito) [reply tox.ini:3]: I think we can fix the pylint issues in a specific PR

**Comment by francescofuggitti** [comment]
LGTM! We just need to fix the new test case failing.

**Pull request merged**